optuna/search_space/intersection.py

```diff
@@ -1,13 +1,55 @@
+from __future__ import annotations
+
 from collections import OrderedDict
 import copy
 from typing import Dict
-from typing import Optional
+from typing import List
```

Comment:
Since we introduce `__futuer__.annotations`, we can use `list` instead of  `typing.List`.
```suggestion
```

Reply:
`typing.Tuple` is still required according to https://github.com/optuna/optuna/pull/4526/files#r1148474747.

Reply:
Actually, it doesn't.

Reply:
What is unsupported on < python3.10 is not the `Tuple` but the `TypeAlias` feature. TypeAlian cannot be used with the `|` operator.

Reply:
Thank you for your suggestions, But, can I leave this topic to a follow-up PR?